### PR TITLE
feat: SMS and email OTP texts

### PR DESCRIPTION
**Pull request by eliobischof**
### Console
- [ ] Makes the two message texts for SMS and email OTP verification customizable via API and console.
- [ ] Omits the buttonText property in the console for both the new SMS OTP verification and the existing phone verification texts.
- [ ] Fixes a console bug for resetting instance level message texts.

### API
- [ ] Manage SMS OTP and Email OTP via API
- [ ] Omits the buttonText property in the API for the new SMS OTP verification text.

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [x] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

**Review by S-code652 — APPROVED** [review]

File: proto/zitadel/admin.proto [hunk 1]
```diff
@@ -2953,7 +2953,7 @@ service AdminService {
         };
 
         option (zitadel.v1.auth_option) = {
-            permission: "iam.policy.delete"
+            permission: "iam.policy.delete" // TODO: iam.policy.write on all reset text endpoints?
```
Comment (livio-a) [inline proto/zitadel/admin.proto:2956]: ?

File: internal/notification/static/i18n/zh.yaml [hunk 2]
```diff
@@ -26,13 +26,19 @@ VerifyPhone:
...
+  Greeting: 你好，{{.FirstName}} {{.LastName}}，
+  Text: 请使用 '验证' 按钮，或复制一次性密码 {{.OTP}} 并将其粘贴到验证屏幕中，以在接下来的五分钟内在 ZITADEL 中进行验证。
+  ButtonText: 验证
 DomainClaimed:
   Title: ZITADEL - 域名所有权验证
   PreHeader: 更改电子邮件/用户名
   Subject: 域名所有权验证
   Greeting: 你好 {{.DisplayName}},
   Text: 域 {{.Domain}} 已被组织使用。您当前的用户 {{.Username}} 不属于此组织。因此，您必须在登录时更改您的电子邮件。我们为此登录创建了一个临时用户名 ({{.TempUsername}})。
-  ButtonText: 登录
```
Comment (livio-a) [inline internal/notification/static/i18n/zh.yaml:35]: why is that removed? and where is the sms text?

File: internal/notification/static/i18n/de.yaml [hunk 3]
```diff
@@ -26,13 +26,25 @@ VerifyPhone:
...
+  Subject: Einmalpasswort verifizieren
+  Greeting: Hallo {{.FirstName}} {{.LastName}},
+  Text: Bitte besuche {{ .VerifyURL }} oder kopiere das Einmalpasswort {{.OTP}} und füge es in den Authentifizierungsbildschirm ein, um dich innerhalb der nächsten fünf Minuten bei ZITADEL zu authentifizieren.
 DomainClaimed:
   Title: ZITADEL - Domain wurde beansprucht
   PreHeader: Email / Username ändern
   Subject: Domain wurde beansprucht
   Greeting: Hallo {{.DisplayName}},
   Text: Die Domain {{.Domain}} wurde von einer Organisation beansprucht. Dein derzeitiger User {{.Username}} ist nicht Teil dieser Organisation. Daher musst du beim nächsten Login eine neue Email hinterlegen. Für diesen Login haben wir dir einen temporären Usernamen ({{.TempUsername}}) erstellt.
-  ButtonText: Login
```
Comment (livio-a) [inline internal/notification/static/i18n/de.yaml:35]: why is this removed?

File: internal/i18n/i18n.go [hunk 4]
```diff
@@ -13,7 +13,7 @@ import (
 	"github.com/nicksnyder/go-i18n/v2/i18n"
 	"github.com/zitadel/logging"
 	"golang.org/x/text/language"
-	"sigs.k8s.io/yaml"
+	"sigs.k8s.io/yaml" // TODO: not gopkg.in/yaml.v3?
```
Comment (livio-a) [inline internal/i18n/i18n.go:16]: ?

**Review by livio-a — CHANGES_REQUESTED** [review]
i only checked the backend, but there are a lot of TODOs
please also note that (in preparation of making them not required anymore) we changed the `{{.FirstName}} {{.LastName}}` combination to {{.DisplayName}}

File: internal/notification/static/i18n/zh.yaml (hunk 2 above)
Comment (eliobischof) [reply internal/notification/static/i18n/zh.yaml:35]: Can we render buttons in SMS?

File: proto/zitadel/admin.proto (hunk 1 above)
Comment (eliobischof) [reply proto/zitadel/admin.proto:2956]: Why is it not iam.policy.write?

File: internal/i18n/i18n.go (hunk 4 above)
Comment (eliobischof) [reply internal/i18n/i18n.go:16]: Isit intended to use the k8s yaml library?

File: internal/notification/static/i18n/zh.yaml (hunk 2 above)
Comment (eliobischof) [reply internal/notification/static/i18n/zh.yaml:35]: LOL, idk

Comment (eliobischof) [reply internal/notification/static/i18n/zh.yaml]: readded these translations

**Comment by eliobischof** [comment]
> please also note that (in preparation of making them not required anymore) we changed the `{{.FirstName}} {{.LastName}}` combination to {{.DisplayName}}

I fixed the translations to use DisplayName now.

> i only checked the backend, but there are a lot of TODOs 

Yes, these are questions to you in person :smile: Can you answer them? I'll then create follow-ups or delete the TODOs.

File: internal/notification/static/i18n/de.yaml (hunk 3 above)
Comment (eliobischof) [reply internal/notification/static/i18n/de.yaml]: fixed

**Comment by peintnermax** [comment]
@eliobischof there are some translations missing for:
```
 "VSO": "Verify SMS OTP",
 "VEO": "Verify Email OTP",
```
in all but en.json. The rest looks good IMO

**Comment by eliobischof** [comment]
> @eliobischof there are some translations missing for:
> 
> ```
>  "VSO": "Verify SMS OTP",
>  "VEO": "Verify Email OTP",
> ```
> 
> in all but en.json. The rest looks good IMO

done, thanks

**Comment by eliobischof** [comment]
@livio-a I removed all TODO comments and the pointless sms text properties. Can you review again, plz?

File: internal/notification/static/i18n/en.yaml [hunk 5]
```diff
@@ -26,6 +26,15 @@ VerifyPhone:
...
   ButtonText: Verify phone
+VerifyEmailOTP:
+  Title: ZITADEL - Verify One-Time Password
+  PreHeader: Verify One-Time Password
+  Subject: Verify One-Time Password
+  Greeting: Hello {{.DisplayName}},
+  Text: Please use the "Authenticate" button or copy the one-time password {{.OTP}} and paste it to to the authentication screen in order to authenticate at ZITADEL within the next five minutes.
+  ButtonText: Authenticate
+VerifySMSOTP:
+  Text: Please visit {{ .VerifyURL }} or copy the one-time password {{.OTP}} and paste it to to the authentication screen in order to authenticate at ZITADEL within the next five minutes.
```
Comment (eliobischof) [inline internal/notification/static/i18n/en.yaml:37]: @livio-a is this default text correct?

Comment (livio-a) [reply internal/notification/static/i18n/en.yaml:37]: LGTM

**Comment by livio-a** [comment]
if checked out the current state and run it locally
LGTM, but e.g. at `Verify SMS OTP` there's a `details` and should we hide the texts, which cannot be set?
image

cc @peintnermax

**Comment by eliobischof** [comment]
> if checked out the current state and run it locally
> LGTM, but e.g. at `Verify SMS OTP` there's a `details` and should we hide the texts, which cannot be set?
> image
> 
> cc @peintnermax 

Did you rebuild the console?

**Comment by livio-a** [comment]
> 
> Did you rebuild the console?

i used make and probably missed a step to embed it into my local binary. just retried it and now it works, sorry.

**Review by livio-a — APPROVED** [review]
